modules/frontend/search_sharder.go

```diff
@@ -91,6 +92,10 @@ func (s asyncSearchSharder) RoundTrip(pipelineRequest pipeline.Request) (pipelin
 		return pipeline.NewBadRequest(fmt.Errorf("range specified by start and end exceeds %s. received start=%d end=%d", maxDuration, searchReq.Start, searchReq.End)), nil
 	}
 
+	if searchReq.SpansPerSpanSet > s.cfg.MaxSpansPerSpanSet {
+		return pipeline.NewBadRequest(fmt.Errorf("spans per span set exceeds %d. received %d", s.cfg.MaxSpansPerSpanSet, searchReq.SpansPerSpanSet)), nil
```

Comment:
Not sure how to call it in the error message because in the Web UI the field is called `Span Limit` while in the API it is called `spss`.

Reply:
that's a good call out. i think calling it "spans per spanset" is fine here. that's what i would do.